### ISSUE TITLE
macOS installation fails

### ISSUE THREAD

**Issue by unphased** [ranger/ranger#1926]
#### Runtime Environment

- Operating system and version: macOS Catalina 10.15.4
- Python version: system 2.7.16 & homebrew 3.7.7
- Ranger version/commit: 081e731

#### Current Behavior

[install_log.txt](https://github.com/ranger/ranger/files/4502425/install_log.txt)
`make install` places everything into `usr/local/lib/python2.7/site-packages/ranger`, which appears to be a poor choice given that system python2 is not going to be looking there for its modules: 

```
❯❯❯ python -m site
sys.path = [
    '/Users/slu/util',
    '/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python27.zip',
    '/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7',
    '/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/plat-darwin',
    '/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/plat-mac',
    '/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/plat-mac/lib-scriptpackages',
    '/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/lib-tk',
    '/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/lib-old',
    '/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/lib-dynload',
    '/Library/Python/2.7/site-packages',
    '/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python',
    '/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/PyObjC',
]
USER_BASE: '/Users/slu/Library/Python/2.7' (doesn't exist)
USER_SITE: '/Users/slu/Library/Python/2.7/lib/python/site-packages' (doesn't exist)
ENABLE_USER_SITE: True
```

#### Expected Behavior
not to see 
```
Traceback (most recent call last):
  File "/usr/local/bin/ranger", line 36, in <module>
    import ranger  # NOQA pylint: disable=import-self,wrong-import-position
ImportError: No module named ranger
```
upon running ranger.

#### Possible Solutions
- instructions amended for the user to know ahead of time that they need to point the installer  to where modules will be searched. And/or perhaps some sort of guidance on python 2 vs python 3. 

Alternatively, make the makefile smarter and do the right thing.

#### Steps to reproduce
1. $ git clone git@github.com:ranger/ranger.git
2. $ make install
3. $ ranger

**Comment by unphased** [comment]
I'll admit that since I did not test this on a fresh machine it may be possible that something weird in my setup is causing the `make install` to misbehave like this.

**Comment by unphased** [comment]
Another note, the contents of `install_log.txt` are missing the leading slash.

**Comment by toonn** [comment]
Thanks for the report. I guess we should check the PYTHONPATH and complain if the chosen PREFIX isn't in there.

Why not install ranger using homebrew, btw?

**Comment by unphased** [comment]
I was using it from brew like a normal person before. And was observing some rather frustrating freezing with previews. It looks like the `ranger/data/scope.sh` is hanging. When I terminate it, interactivity in ranger is restored. 

I was blindly hoping running the latest would make this problem go away, but it has not. So I'm still digging around on that.

**Comment by toonn** [comment]
I've noticed this on macos as well.

**Comment by rajasegar-c** [comment]
I am also facing issues while installing through Homebrew

```
 brew install ranger
Updating Homebrew...
==> Downloading https://ranger.github.io/ranger-1.9.3.tar.gz
Already downloaded: /Users/user/Library/Caches/Homebrew/downloads/44b65ede52870f573cbcd456934c6de9db1bb08f130d130e79c09ee5e2d4040a--ranger-1.9.3.tar.gz
Error: An exception occurred within a child process:
  NoMethodError: undefined method `path' for nil:NilClass
Did you mean?  paths
```

**Comment by toonn** [comment]
@rajasegar-c, that doesn't look like a ranger issue. Contact either the ranger formula maintainer or the homebrew team.

**Comment by rajasegar-c** [comment]
Thanks @toonn I will check with them

**Comment by thrsam005** [comment]
> I was indeed using it from brew like a normal person before. And was observing some rather frustrating freezing with previews. It looks like the `ranger/data/scope.sh` is hanging. When I terminate it, interactivity in ranger is restored.
> 
> I was blindly hoping running the latest would make this problem go away, but it has not. So I'm still digging around on that.

Did you ever figure out a permanent solution for this issue? I am on Big Sur experiencing this same issue, and I was also experiencing it on Catalina. I even switched to an older machine running High Sierra because it was so frustrating for me, and had no issues with freezing previews.

**Comment by toonn** [comment]
I run into this on High Sierra regularly. I don't know what causes it and I haven't had time to look into it. It'd be great if someone could look into this.

**Comment by thrsam005** [comment]
I never had any issues running ranger in High Sierra on my 2010 iMac - it did take quite a bit of work to get it to run properly, and IIRC I used MacPorts to install ranger (and vim) instead of homebrew because I was running into issues where vim and ranger would only recognise my python2 install and not my python3 install. 

I did the same on my 2017 MacBook Pro but ranger still freezes sometimes, so I'm at a loss right now.

**Comment by toonn** [comment]
I can't imagine it makes a difference but you could also try installing `ranger-fm` via pip.

**Comment by thrsam005** [comment]
@toonn no, that did not make a difference. When ranger freezes, all I am able to do is quit terminal and re-launch. It gets stuck and will not respond in any reasonable amount of time. 

Another thing that happens that I just noticed is that if I'm using ranger to browse through files, and then I decide to quit ranger (via :q), ranger will not allow me to quit due to "tasks in progress" - these tasks, whatever they are, will not terminate in any reasonable amount of time and so I have to force ranger to quit. Does anything like this happen to you?

This is making me re-think my choice of file manager.

**Comment by toonn** [comment]
Yes, this happens to me too. ^C always stops the tasks and lets me continue using ranger though. I don't know why this happens though. Somehow on macOS the preview processes are a *lot* slower than on linux.

**Comment by thrsam005** [comment]
Again this is just so strange, I don't know how I was able to use ranger flawlessly on High Sierra (for about a year) while you've had issues on the same OS. There must be something we're missing.

**Comment by toonn** [comment]
I think the next step would be to "bisect" `scope.sh` to figure out whether there's a specific program that triggers this. So, comment out ~half of the remaining programs until the issue doesn't turn up anymore then switch which half is commented out until it's narrowed down to a single (hopefully) misbehaving program. I suspect image previews.

**Comment by thrsam005** [comment]
@toonn I made some progress. I went back and found that there exists a ranger port for MacPorts. I installed ranger through MacPorts and it seems to have the same issues - not that I'm particularly surprised, but I had hoped that if a port existed that people had worked out the issue (whether it's from ```scope.sh``` or not, why it does these things, etc.) so that's a bit disappointing. 

I did take a look at ```scope.sh``` and decided modifying that file is above my pay grade, hence I just opted to put ```set user_preview_script false``` into my ```rc.conf``` file as you suggested. I will eventually dig into this further but work is very busy at the moment so I won't get the chance for a while.
